packages/components/src/PieChartButton/PieChartButton.component.js

```diff
@@ -0,0 +1,387 @@
+		if (amountToSubtract > 0) {
+			// eslint-disable-next-line no-param-reassign
+			element.percentageShown -= 1;
+			amountToSubtract -= 1;
+		}
+	}
+
+	while (amountToSubtract > 0) {
+		const elementsToDecrease = valuesMins
+			.filter(value => value.percentageShown > minimumPercentage)
+			.sort(sortElements);
+		if (elementsToDecrease.length > 0) {
+			elementsToDecrease.forEach(decreaseElement);
+		} else {
+			amountToSubtract = 0;
+		}
+	}
+
+	return valuesMins;
+}
+
+/**
+ * This function add an OverlayTrigger wrapping the button if defined
+ * @param {Element} btn the current button
+ * @param {string} overlayPlacement the overlay placement
+ * @param {Element} overlayComponent the overlay component
+ * @param {string} overlayId the id to be set for the overlay
+ */
+export function decorateWithOverlay(btn, overlayPlacement, overlayComponent, overlayId) {
+	if (!overlayComponent) {
+		return btn;
+	}
+
+	return (
+		// this span is here to allow the tooltip trigger to work
+		<span>
+			<OverlayTrigger
+				trigger="click"
+				rootClose
+				placement={overlayPlacement}
+				overlay={<Popover id={overlayId}>{overlayComponent}</Popover>}
+			>
+				{btn}
+			</OverlayTrigger>
+		</span>
+	);
+}
+
+/**
+ * This function wrap the button with a TooltipTrigger
+ * @param {Element} btn the button element ( may be wrapped by overlay trigger )
+ * @param {boolean} tooltip tell if the tooltip has to be showed
+ * @param {string} label the label to show on the tooltip
+ * @param {string} tooltipPlacement the tooltip placement
+ */
+export function decorateWithTooltip(btn, tooltip, label, tooltipPlacement) {
+	if (!tooltip || !label) {
+		return btn;
+	}
+	return (
+		<TooltipTrigger label={label} tooltipPlacement={tooltipPlacement}>
+			{btn}
+		</TooltipTrigger>
+	);
+}
+
+/**
+ * This function wrap the event when we don't have an overlay component
+ * @param {function} mouseEvent the event to wrap ( mouseClick )
+ * @param {Element} overlayComponent tell if there is an overlay component
+ * @param {string} label the label of the component
+ * @param {object} rest the rest of the props
+ * @param {object} model the model of the component
+ */
+export function wrapMouseEvent(mouseEvent, overlayComponent, label, rest, model) {
+	if (overlayComponent || !mouseEvent) {
+		return null;
+	}
+	return event =>
+		mouseEvent(event, {
+			action: { label, ...rest },
+			model,
+		});
+}
+
+/**
+ * This function return useful stuff to build the graph of the loader
+ * @param {number} size the size in px of the graph
+ */
+export function getDisplaySize(size) {
+	const innerRadiusPerPixel = 12 / 30;
+	const outerRadiusPerPixel = 14 / 30;
+	const padAnglePerPixel = 0.1 / 30;
+
+	const pixelNumber = size - 20;
+	return {
+		svgSize: size,
+		innerRadius: parseInt(6 + innerRadiusPerPixel * pixelNumber, 10),
+		outerRadius: parseInt(9 + outerRadiusPerPixel * pixelNumber, 10),
+		padAngle: 0.2 - padAnglePerPixel * pixelNumber,
+	};
```

Comment:
Looks like a lot of those values are arbitrary. They should be extracted into constant with proper names.